### PR TITLE
Fixes strict compilation errors.

### PR DESCRIPTION
**Pull request by tommyzed**
## Description

Fixed strict compilation errors.

1) Cast error.code to long
2) Changes return "bool" to "BOOL"

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_messaging/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 5.1.7
```
[the remark this answers is not in the document]
Comment (tommyzed) [reply packages/firebase_messaging/CHANGELOG.md]: @mehmetf -- Can I add a file to an existing pull request or do I need to create a new one?

**Comment by tommyzed** [comment]
How can these checks be re-run? The error doesn't seem to have anything to do with my CL...

packages/flutter_tools/lib/src/build_runner/resident_web_runner.dart:483:38: Error: Method not found: 'ProjectFileInvalidator.findInvalidated'.
        await ProjectFileInvalidator.findInvalidated(
                                     ^^^^^^^^^^^^^^^

**Comment by tommyzed** [comment]
> @tommyzed Thanks for the PR. I will have a look at this once #121 has been merged in.

@kroikie --> Hi, what can I do to push this in? I'm not understanding why this check is failing...

**Comment by mehmetf** [comment]
Folks I am going to merge this. The test that's failing looks flaky and it helps us cleanup internal lints.

**Pull request merged**